### ISSUE TITLE
Memory leak when setting nested keys using dot notation

### ISSUE THREAD

**Issue by raducorcan** [jenssegers/laravel-mongodb#2959]
- Laravel-mongodb Version: 4.3.0
- PHP Version: 8.3
- Database Driver & Version: 1.15.0 & 5.0

### Description:

There appears to be a memory leak when setting nested keys using dot notation. I am attaching a very simple code snippet that exemplifies the issue.

### Steps to reproduce
```
class A extends \MongoDB\Laravel\Eloquent\Model {};

$a = new A;
for ($i = 0; $i < 100; $i++) {
  $a['somerandomkey'] = 42;
  echo (memory_get_usage() / (1024*1024) . "\n"); // this prints the same value in every iteration
}

//----------------------------------

class A extends \MongoDB\Laravel\Eloquent\Model {};

$a = new A;
for ($i = 0; $i < 100; $i++) {
  $a['some.random.key'] = 42;
  echo (memory_get_usage() / (1024*1024) . "\n"); // this prints increasingly larger values
}

```

### Expected behaviour
Script memory usage keeps increasing.

### Actual behaviour
The memory usage should remain constant.

**Comment by GromNaN** [comment]
Thanks for the report. I can confirm this is specific to the MongoDB package as there is no memory leak by extending `Illuminate\Database\Eloquent\Model`.

Tracking in Jira for investigation: PHPORM-184

**Comment by GromNaN** [comment]
There is a hack in `Model::setAttribute` that uses a temporary key and then remove it. I suspect the memory leak to come from here. https://github.com/mongodb/laravel-mongodb/blob/b892bb089c25c84c602318dadae59f89bf8dff5b/src/Eloquent/Model.php#L272-L282

**Comment by GromNaN** [comment]
Fixed by #2962
